aea/configurations/data_types.py

```diff
@@ -284,17 +290,24 @@ def package_version(self) -> PackageVersion:
         """Get the package version object."""
         return self._package_version
 
-    def to_any(self) -> "PublicId":
-        """Return the same public id, but with any version."""
-        return PublicId(self.author, self.name, self.ANY_VERSION)
+    @property
+    def hash(self,) -> str:
+        """Returns the hash for the package."""
+        if self._package_hash is None:
+            raise ValueError("Package hash was not provided.")
+        return str(self._package_hash)
 
     def same_prefix(self, other: "PublicId") -> bool:
         """Check if the other public id has the same author and name of this."""
         return self.name == other.name and self.author == other.author
 
+    def to_any(self) -> "PublicId":
```

Comment:
This is not backwards compatible.

Why not add a new method `to_any_with_hash`?